elemica-suggest.coffee

```diff
@@ -154,10 +160,11 @@ elemicaSuggest 0.7.1-SNAPSHOT.
         $(this).attr 'autocomplete', 'off'
 
         $(this).on 'blur', (event) ->
-          removeSuggestions event.target
+          $target = $(event.target)
+          removeSuggestions $target
 
           if $valueInput.val() == ""
-            $(event.target).val("")
+            $target.val("") if noSuggestionMatched($target.val())
             afterSelect(null)
```

Comment:
On the other hand - should we invoke `afterSelect` in this case at all? Nothing was selected actually. Maybe we should drop this `afterSelect(null)` invocation and if user wants to run this code he should be responsible for invoking it manually.